chctools/btor2.py

```diff
@@ -357,9 +357,12 @@ def to_fresh_z3_expr(self, m: MutableMapping[int, z3.ExprRef]) -> z3.BoolRef:
 
 
 class Redxor(BitvecUnaryOp):
-    def to_fresh_z3_expr(self, m: MutableMapping[int, z3.ExprRef]) -> z3.BoolRef:
-        # TODO
-        raise ValueError
+    def to_fresh_z3_expr(self, m: MutableMapping[int, z3.ExprRef]) -> z3.BitVecRef:
+        child_z3_bitvec: z3.BitVecRef = self.get_child_z3_bitvec(m)
+        z3_expr: z3.BitVecRef = z3.Extract(0, 0, child_z3_bitvec)
+        for i in range(1, self.width()):
+            z3_expr = z3_expr + z3.Extract(i, i, child_z3_bitvec)
```

Comment:
is this really xor?

Reply:
The [Btor2 paper](https://link.springer.com/chapter/10.1007/978-3-319-96145-3_32) says that `redxor` is a unary operator that takes a `bitvec` and returns a `bitvec` of width 1, but it doesn't define the semantics of it. The [Btor paper](https://www.researchgate.net/publication/234826506_BTOR_bit-precise_modelling_of_word-level_problems_for_model_checking) says that `redxor` it is the same as Verilog's `red_xor` operator. I have found [this page](https://books.google.ca/books?id=QGlQDwAAQBAJ&pg=PA74&lpg=PA74&dq=Verilog+redxor&source=bl&ots=Xo1f9clml3&sig=ACfU3U1qJ5aQKsmnvRNu5USO5QMLvnhw2Q&hl=en&sa=X&ved=2ahUKEwiCvfXC5qPsAhXVHc0KHUnyDtkQ6AEwBHoECAEQAg#v=onepage&q=Verilog%20redxor&f=false) on Google Books and it seems like `redxor` returns the parity of a `bitvec`  (i.e. returns 1 if there are an odd number of ones and return 0 otherwise).

Reply:
The only way that I can think of to implement `redxor` is to use `z3.Extract` to get each bit of the `bitvec`, and then use `bvxor` repeatedly on each pair of bits until there is only one bit left.

Reply:
I was not sure that `+` is `bvxor`